### ISSUE TITLE
Http request connectors with managed identity are not working locally (Logic App Standard)

### ISSUE THREAD

**Issue by timjansenortec** [Azure/logicapps#580]
I am using a Http request with ManagedServiceIdentity to get a secret from a keyvault:
"Some_Action": {
    "inputs": {
        "authentication": {
            "audience": "https://vault.azure.net",
            "type": "ManagedServiceIdentity"
        },
        "method": "GET",
        "uri": "@{parameters('keyVaultUrl')}/secrets/SomeSecret?api-version=7.1"
    },
    "runAfter": {},
    "type": "Http"
},

This gives the following error when running locally in Visual Studio Code:
'The operation 'Some_Action' for workflow 'XX' contains managed identity authentication which is supported only for Logic Apps Standard or Logic Apps running in App Service Environments.

Am I doing something wrong?
Is this even a supported scenario?
It would be nice if logic apps could get the identity from the context, the same way that DefaultAzureCredential does in a coded app (https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet).

AB#16099601

**Comment by bydynamics** [comment]
Same issue, question. are managed identities (user) supported for the vs code local scenario? 

[2022-07-26T15:47:34.411Z] Workflow Error: operationName='WorkflowDefinitionProvider.ProcessWorkflow', message='Workflow 'sb-employees_bc-employees-resources' validate and create workflow operation failed, the exception is 'The operation 'HTTPRunQueryAndListResultsTranslations' for workflow
'sb-employees_bc-employees-resources' contains managed identity authentication which is supported only for Logic Apps Standard or Logic Apps running in App Service Environments.'', exception='Microsoft.Azure.Workflows.Common.ErrorResponses.ErrorResponseMessageException: The operation 'HTTPRunQueryAndListResultsTranslations' for workflow 'sb-employees_bc-employees-resources' contains managed identity authentication which is supported only for Logic Apps Standard or Logic Apps running in App Service Environments

**Comment by timjansenortec** [comment]
As a workaround, I have written my own custom connector that internally uses DefaultAzureCredentials. This works fine. For me this is worth the effort, since I am doing a bunch of other things in the custom connector. For the most basic scenario, this is way too hassle.

In my opinion, using ManagedServiceIdentity should 'just' work when running locally in vs code.

**Comment by StefanPuntNL** [comment]
Same issue and question. @derek1ee Could you shed some light on this scenario?

**Comment by kseeman** [comment]
I'm also running into the same issue. Was anyone able to find a solution or is creating a custom connector the only work-around?

**Comment by rohithah** [comment]
managed identity is not supported when running locally. You need to either run this in azure or you can configure AAD authentication on the action either with clientId/Secret or a certificate

**Comment by simonp92** [comment]
Is support for managed identities in VSCode on the roadmap?

It would enhance the local development experience quite a bit

**Comment by yoHasse** [comment]
@rohithah 
So how are we supposed to work with Logic Apps Standard if a main security feature is not available locally?
Do you have any suggestions on how locally work with connection strings/certificates etc. and the during deployment switch to Managed Identity?

Examples:
* Can we create a resouce that can be used in the cloud and switch by using appsettings?
* Can we create a connection JSON that can be switched using appsettings?
* Can we create create a connections.local.json that uses connection string and then having a main connections.json that use managed identity?

Our main point for using Logic App Standard was the option to work locally but we must still comply with our security Team.
Manually switching when building locally vs in the cloud should not be an option.

**Comment by mark-abrams** [comment]
This issue with Managed Identity authentication when developing locally is essentially the same issue that I raised regarding the built-in Service Bus connector and Managed Identity - #708. Like you, I was told that Managed Identity is not supported for built-in connectors when running locally. And like you, this limitation is preventing us from creating any Logic Apps using Managed Identity. I don't understand why this limitation exists when Azure Functions running locally have supported Managed Identity authentication for triggers and bindings for ages.

I have also created discussion #715 where I have have requested Managed Identity support for the Service bus connector when running locally, and also asked that this is supported for all built-in connectors that can use Managed Identity. I've not had any reply.
